app/handlers/sessions_create.rb

```diff
@@ -57,7 +57,14 @@ def handle
         status = :authentication_taken
       else
         return outputs[:status] = :new_signin_required if user_signin_is_too_old?
-        run(TransferAuthentications, authentication, current_user)
+        auth_transfer = run(TransferAuthentications, authentication, current_user)
+
+        if auth_transfer.outputs[:same_provider]
+          outputs[:status] = :same_provider
+          outputs[:authentication] = authentication
```

Comment:
If need it, why not just set it up at the top so all callers can access it?

```
outputs[:authentication] = authentication = 
  Authentication.find_or_create_...
```

This way you can get it back even if you follow my suggestion above to `return outputs[:status] = ...`